frontend/src/components/DefenceBox/DefenceConfigurationInput.tsx

```diff
@@ -34,40 +57,62 @@ function DefenceConfigurationInput({
 		// shift+enter shouldn't send message
 		if (event.key === 'Enter' && !event.shiftKey) {
 			// asynchronously send the message
-			setConfigurationValue(value);
+			void setConfigurationValueIfDifferent(value);
+		}
+	}
+
+	function validateNewInput(value: string) {
+		if (validateDefence(defence.id, config.id, value)) {
+			setConfigValidated(true);
+		} else {
+			setConfigValidated(false);
 		}
 	}
 
 	if (inputType === 'text') {
 		return (
-			<ThemedTextArea
-				id={id}
-				content={value}
-				onContentChanged={setValue}
-				disabled={disabled}
-				maxLines={10}
-				onBlur={() => {
-					setConfigurationValue(value);
-				}}
-				onKeyDown={inputKeyDown}
-				onKeyUp={inputKeyUp}
-				characterLimit={CONFIG_VALUE_CHARACTER_LIMIT}
-			/>
+			<>
+				<ThemedTextArea
+					id={id}
+					content={value}
+					onContentChanged={setValue}
+					configValidated={configValidated}
+					validateInput={validateNewInput}
+					disabled={disabled}
+					maxLines={10}
+					onBlur={() => {
+						void setConfigurationValueIfDifferent(value);
+					}}
+					onKeyDown={inputKeyDown}
+					onKeyUp={inputKeyUp}
+					characterLimit={CONFIG_VALUE_CHARACTER_LIMIT}
+				/>
+				<p className="error-message" aria-live="polite">
+					{configValidated ? '' : 'Error: Invalid configuration value'}
+				</p>
```

Comment:
You have this error JSX duplicated now. Better to declare your input separately, then return it in a fragment:

```
const inputElement = (inputType === 'text)
  ? (
    <ThemedTextArea ... />
  ) : (
    <ThemedNumberInput ... />
  );

return (
  <>
    {inputElement}
    {!configValidated && (
      <p className="error-message" aria-live="polite"> ... </p>
    )}
  </>
);
```